src/info/info.c

```diff
@@ -24,358 +24,412 @@ Info Handler
+/***********************************************************************************************************************************
+Save a value and update checksum
+***********************************************************************************************************************************/
+void
+infoSaveValue(InfoSave *infoSaveData, const String *section, const String *key, const String *value)
+{
+    FUNCTION_TEST_BEGIN();
+        FUNCTION_TEST_PARAM(INFO_SAVE, infoSaveData);
+        FUNCTION_TEST_PARAM(STRING, section);
+        FUNCTION_TEST_PARAM(STRING, key);
+        FUNCTION_TEST_PARAM(STRING, value);
+    FUNCTION_TEST_END();
 
-    Info *this = infoNewInternal();
+    ASSERT(infoSaveData != NULL);
+    ASSERT(section != NULL);
+    ASSERT(key != NULL);
+    ASSERT(value != NULL);
 
-    MEM_CONTEXT_BEGIN(this->memContext)
+    // Save section
+    if (infoSaveData->sectionLast == NULL || !strEq(section, infoSaveData->sectionLast))
     {
-        Ini *iniLocal = NULL;
-
-        // Attempt to load the primary file
-        TRY_BEGIN()
+        if (infoSaveData->sectionLast != NULL)
         {
-            iniLocal = infoLoad(this, storage, fileName, false, cipherType, cipherPass);
+            INFO_CHECKSUM_SECTION_NEXT(infoSaveData->checksum);
+            ioWriteLine(infoSaveData->write, BUFSTRDEF(""));
         }
-        CATCH_ANY()
-        {
-            // On error store the error and try to load the copy
-            String *primaryError = strNewFmt("%s: %s", errorTypeName(errorType()), errorMessage());
-            bool primaryMissing = errorType() == &FileMissingError;
-            const ErrorType *primaryErrorType = errorType();
 
-            TRY_BEGIN()
-            {
-                iniLocal = infoLoad(this, storage, fileName, true, cipherType, cipherPass);
-            }
-            CATCH_ANY()
-            {
-                // If both copies of the file have the same error then throw that error,
-                // else if one file is missing but the other is in error and it is not missing, throw that error
-                // else throw an open error
-                THROWP_FMT(
-                    errorType() == primaryErrorType ? errorType() :
-                        (errorType() == &FileMissingError ? primaryErrorType :
-                        (primaryMissing ? errorType() : &FileOpenError)),
-                    "unable to load info file '%s' or '%s" INFO_COPY_EXT "':\n%s\n%s: %s",
-                    strPtr(storagePathNP(storage, fileName)), strPtr(storagePathNP(storage, fileName)),
-                    strPtr(primaryError), errorTypeName(errorType()), errorMessage());
-            }
-            TRY_END();
-        }
-        TRY_END();
-
-        // Load the cipher passphrase if it exists
-        const String *cipherPassSub = iniGetDefault(iniLocal, INFO_SECTION_CIPHER_STR, INFO_KEY_CIPHER_PASS_STR, NULL);
+        INFO_CHECKSUM_SECTION(infoSaveData->checksum, section);
```

Comment:
Sort of.  The `infoWrite` calls are writing out an ini file, but the checksum macros are writing a JSON object because that how it was done in Perl and we can't change how the checksums are calculated.